samples/notebooks/python/07-native-function-inline.ipynb

```diff
@@ -0,0 +1,510 @@
+{
+  "cells": [
+    {
+      "attachments": {},
+      "cell_type": "markdown",
+      "id": "3c93ac5b",
+      "metadata": {},
+      "source": [
+        "# Running Native Functions"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "id": "ebcabb91",
+      "metadata": {},
+      "source": []
+    },
+    {
+      "attachments": {},
+      "cell_type": "markdown",
+      "id": "40201641",
+      "metadata": {},
+      "source": [
+        "Two of the previous notebooks showed how to [execute semantic functions inline](./03-running-semantic-function-inline.ipynb) and how to [run prompts from a file](./02-running-prompts-from-file.ipynb).\n",
+        "\n",
+        "In this notebook, we'll show how to use native functions from a file. We will also show how to call semantic functions from native functions.\n",
```

Comment:
So in addition to this, one thing that we need to do a better job showing is how to within the prompt of a semantic function, how we can mix and call other native and semantic functions from various skills. 

Right now, the only example showing this is memory notebook calling the TextMemory skill's recall function and it's not the most obvious to people that that's actually a native function being called inside this new semantic function. 

Could you extend the examples you have here to spell that out more? Let me know if you need more detail!

Reply:
Making use of some of the Semantic Kernel's already built-in core skills could be good.

Reply:
And also highlighting and explaining the `@sk_function_context_parameter` decorator would be great too

Reply:
So overall in the notebooks, we want to take a reader through the journey of:

1) How to create semantic skills with prompts. Introduce the prompt templating language
2) How to use context variables to populate values into the prompt
3) How to mix and call Semantic functions and Native Functions within the prompt (this notebook)

Reply:
@alexchaomander I've got most of those changes you requested put in. 
I could use some clarification on the `@sk_function_context_parameter` decorator though. Aside from providing documentation about what needs to be provided in the `ContextVariables`, it doesn't seem to do anything functionally. Am I missing something?

Reply:
So, it's mostly used for Planner so the Kernel will know the argument description. (See https://github.com/microsoft/semantic-kernel/pull/597)

Also from https://github.com/microsoft/semantic-kernel/issues/624, apparently in order to support functions with multiple inputs, you'd need to use the context_parameter decorator

Reply:
Ok the interaction with the planner makes sense, but I did just test this and multiple inputs works with or without the decorator. Not sure if that should be enforced. Seems out of scope for this discussion.

I'll add info about the planner to this doc, that seems helpful.

Reply:
@alexchaomander any other inputs/feedback or can we resolve this conversation?